src/util/trace.h

```diff
@@ -7,6 +7,44 @@
 
 #ifdef ENABLE_TRACING
 
+#ifdef __APPLE__
+
+#include <util/macros.h>
+#include <util/probes.h>
+
+// since the dtrace macros are automatically generated in uppercase, additional
+// macros are needed to translate the lowercase context & event names into the
+// required uppercase CONTEXT_EVENT macros
```

Comment:
we could change that, if this makes it a lot easier for macOS. This would probably be an API break, but I think the tracepoints are still experimental enough to be able to do it.